ext/standard/var.c

```diff
@@ -175,7 +175,16 @@ PHPAPI void php_var_dump(zval *struc, int level) /* {{{ */
 			}
 			ZEND_GUARD_OR_GC_PROTECT_RECURSION(guard, DEBUG, zobj);
 
-			myht = zend_get_properties_for(struc, ZEND_PROP_PURPOSE_DEBUG);
+			// GH-16317: for user classes with __debugInfo() always call that
+			if (ce->type == ZEND_USER_CLASS && ce->__debugInfo != NULL) {
+				int is_temp;
+				myht = (std_object_handlers.get_debug_info)(zobj, &is_temp);
```

Comment:
`get_debug_info` is technically an optional handler.

Reply:
Generally it is optional, but can it ever be unset from the global `std_object_handlers`?

Reply:
Yes, the default handlers may be removed. I don't know exactly why you would do that, but it is currently marked as optional, and its existence is checked for within `zend_std_get_properties_for()`.